# Background

The owner of a company would like to be able to predict whether employees will stay with the company or leave. 

## Data Description

The data contains information about various characteristics of employees. Please note that the dataset has been updated to account for repetitions, which is needed for Goodness of Fit Assessment. See below for the description of these characteristics. 

1. **Age.Group**: 1-9 (1 corresponds to teen, 2 corresponds to twenties, etc.) 
2. **Gender**: 1 if male, 0 if female 
3. **Tenure**: Number of years with the company 
4. **Num.Of.Products**: Number of products owned 
5. **Is.Active.Member**: 1 if active member, 0 if inactive member 
6. **Staying**: Fraction of employees that stayed with the company for a given set of predicting variables.

## Setup

You can import the data and set up the problem with the following R code:

In [1]:
# Import the data
rawdata = read.csv("hw4_data.csv", header=TRUE, fileEncoding="UTF-8-BOM")

# Create variable Staying
rawdata$Staying = rawdata$Stay/rawdata$Employees

# Set variables as categoricals
rawdata$Num.Of.Products<-as.factor(rawdata$Num.Of.Products)
rawdata$Age.Group<-as.factor(rawdata$Age.Group)
rawdata$Gender<-as.factor(rawdata$Gender)
rawdata$Is.Active.Member<-as.factor(rawdata$Is.Active.Member)

# Print head of data
head(rawdata)

Age.Group Gender Tenure Num.Of.Products Is.Active.Member Stay Employees
1 2         1      3      1               0                5    11       
2 2         1      4      1               0                5    10       
3 2         1      4      1               1                2    13       
4 2         0      7      1               0                3    10       
5 2         1      7      1               0                2    14       
6 2         0      4      2               0                4    12       
  Staying  
1 0.4545455
2 0.5000000
3 0.1538462
4 0.3000000
5 0.1428571
6 0.3333333

**Note:** For all of the following questions, treat variables **Tenure** and **Staying** as quantitative variables and **Age.Group**, **Gender**, **Num.Of.Products**, and **Is.Active.Member** as categorical variables. Categorical variables have already been converted to factors in the starter code.

# Question 1: Fitting a Model - 9 pts

Fit a logistic regression model using *Staying* as the response variable with *Num.Of.Products* as the predictor and logit as the link function. Ensure to include the weights parameter for specifying the number of trials. Call it **model1**. Note that *Num.Of.Products* should be treated as a categorical variable.

(a) 3 pts - Display the summary of model1. What are the model parameters and estimates?

In [2]:
model1 <- glm(Staying ~ Num.Of.Products, weights=Employees,family=binomial, data=rawdata)
summary(model1)


Call:
glm(formula = Staying ~ Num.Of.Products, family = binomial, data = rawdata, 
    weights = Employees)

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-4.2827  -1.4676  -0.1022   1.4490   4.7231  

Coefficients:
                 Estimate Std. Error z value Pr(>|z|)    
(Intercept)       0.37886    0.04743   7.988 1.37e-15 ***
Num.Of.Products2 -1.76683    0.10313 -17.132  < 2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 981.04  on 157  degrees of freedom
Residual deviance: 632.04  on 156  degrees of freedom
AIC: 1056.8

Number of Fisher Scoring iterations: 4


(b) 3 pts - Write down the equation for the Odds of Staying.

Odds of Staying = exp(0.379 - 1.767 * Num.Of.Products)

(c) 3 pts - Provide a meaningful interpretation for the estimated coefficient for *Num.Of.Products2* with respect to the log-odds of staying and the odds of staying.

One unit of Num.Of.Products decreases the log odds of staying by 1.767. This corresponds to exp(-1.767) = 0.171 decrease in the odds of staying.

# Question 2: Inference - 9 pts 

(a) 3 pts - Using model1, find a 90% confidence interval for the coefficient for *Num.Of.Products2*.

In [4]:
confint(model1, level = 0.9)

Waiting for profiling to be done...



5 %        95 %      
(Intercept)       0.3010289  0.4570655
Num.Of.Products2 -1.9383610 -1.5989652

The 90% confidence interval for Num.Of.Products2 is (-1.939, -1.599)

In [5]:
1 - pchisq((981.04-632.04), 1)

[1] 0

(b) 3 pts - Is model1 significant overall at the 0.01 significance level?

Yes, the regression is significant overall given the p-value is $\approx 0$

(c) 3 pts - Which regression coefficients are significantly nonzero at the 0.01 significance level? Which are significantly negative? Why?

The intercept and Num.Of.Products2 are statistically significant coefficients at the 0.01 significance level. The only negative coefficient is Num.Of.Products2 which is statistically significant at a 0.01 confidence interval.

# Question 3: Goodness of fit - 10 pts

(a) 3.5 pts - Perform goodness-of-fit hypothesis tests using both Deviance and Pearson residuals. What do you conclude? Explain the differences, if any, between these findings and what you found in Question 2b.

(b) 3.5 pts - Evaluate whether the deviance residuals are normally distributed by producing a QQ plot and histogram of the deviance residuals. What assessments can you make about the goodness of fit of **model1** based on these plots?

(c) 3 pts - Calculate the estimated dispersion parameter for this model. Is this an overdispersed model?

# Question 4: Fitting the full model- 23 pts

Fit a logistic regression model using *Staying* as the response variable with *Age.Group*, *Gender*, *Tenure*, *Num.Of.Products*, and *Is.Active.Member* as the predictors and logit as the link function. Ensure to include the weights parameter for specifying the number of trials. Call it **model2**. Note that Age.Group, Gender, Num.Of.Products, and Is.Active.Member should be treated as categorical variables.

(a) 3 pts - Write down the equation for the probability of staying.

(b) 3 pts - Provide a meaningful interpretation for the estimated coefficients of *Tenure* and *Is.Active.Member1* with respect to the odds of staying.

(c) 3 pts - Is *Is.Active.Member1* statistically significant given the other variables in model2 at the 0.01 significance level?

(d) 10 pts - Has your goodness of fit been affected? Follow the instructions to repeat the tests, plots, and dispersion parameter calculation you performed in Question 3 with **model2**.

(d-1) Perform goodness-of-fit hypothesis tests using both Deviance and Pearson residuals. What do you conclude?

(d-2) Evaluate the linearity assumption of **model2** by plotting the log-odds of Staying vs. **Tenure**. What do you conclude?

(d-3) Evaluate whether the deviance residuals are normally distributed by producing a QQ plot and histogram of the deviance residuals. What do you conclude?

(d-4) Calculate the estimated dispersion parameter for this model. Is this an overdispersed model?

(e) 4 pts - Overall, would you say model2 is a good-fitting model? If so, why? If not, what would you suggest to improve the fit and why? Note: We are not asking you to spend hours finding the best possible model but to offer plausible suggestions along with your reasoning.

# Question 5: Prediction - 9 pts

Suppose there is an employee with the following characteristics:

1. **Age.Group**: 2

2. **Gender**: 0

3. **Tenure**: 2

4. **Num.Of.Products**: 2

5. **Is.Active.Member**: 1

(a) 3 pts - Predict their probability of staying using model1.

(b) 3 pts - Predict their probability of staying using model2.

(c) 3 pts - Comment on how your predictions compare. i.e. which model is more reliable based on the analysis?